### PR TITLE
Added dao class including sql query to retrive job role from database…

### PR DESCRIPTION
**Pull request by Ryan-Jones1999**
… and added model and servie class. Finally I added an endpoint to jobrole class

### CODE REVIEW

**Review by MihneaBocaKainos — APPROVED** [review]

File: src/main/java/com/kainos/ea/model/jobrole.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.kainos.ea.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class jobrole {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/jobrole.java:6]: Class name needs to be upper case.

File: src/main/java/com/kainos/ea/model/jobrole.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    protected int jobid;
+
+    protected String jobName;
+
+    protected String jobResponsibility;
+
+    public int getJobid() {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/jobrole.java:14]: Much of this code is not being used atm. Can you investigate adding Lombok to the project and then just anotating the class accordingly i.e adding @Getter and @Setter tags - https://projectlombok.org/

File: src/main/java/com/kainos/ea/controller/JobApplication.java [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
     }
 
+    @GET
+    @Path("/viewjobroles")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSalesEmployee () {
+        try {
+            return Response.ok(employeeservice.viewJobRoles()).build();
+        } catch (Exception e) {
+            System.out.println(e);
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/JobApplication.java:34]: Remove debugging statement or add a logger

File: src/main/java/com/kainos/ea/controller/JobApplication.java [hunk 4]
```diff
@@ -1,20 +1,40 @@
...
 
     }
 
+    @GET
+    @Path("/viewjobroles")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSalesEmployee () {
+        try {
+            return Response.ok(employeeservice.viewJobRoles()).build();
+        } catch (Exception e) {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/JobApplication.java:33]: Exception is too broad. Masking a runtime Exception. Can you narrow it down to a more relevant exception

File: src/main/java/com/kainos/ea/service/EmployeeService.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    public com.kainos.ea.dao.EmployeeDao EmployeeDao;
+
+    public EmployeeService(EmployeeDao dao){
+        this.EmployeeDao = dao;
+    }
+
+    public List<jobrole> viewJobRoles()   {
+        return EmployeeDao.getjobroles();
+    }
+
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/service/EmployeeService.java:19]: Remove extra lines

File: src/main/java/com/kainos/ea/util/DatabaseConnector.java [hunk 6]
```diff
@@ -33,10 +34,26 @@ public Connection getConnection() throws DatabaseException, SQLException {
...
             return conn;
         } catch (Exception e) {
-            throw new DatabaseException(e);
+            e.printStackTrace();
+            closeConnection();
+        }
+        return null;
+    }
+
+
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/util/DatabaseConnector.java:46]: Remove extra lines

File: src/main/java/com/kainos/ea/model/JobRole.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package com.kainos.ea.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter @Setter
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/JobRole.java:8]: Put setter on new line, for instance 
@Getter
 @Setter
public class SomeClass{..}

File: src/main/java/com/kainos/ea/model/JobRole.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+public class JobRole {
+
+    protected int jobid;
+
+
+    protected String jobName;
+
+
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/JobRole.java:16]: remove extra line

File: src/main/java/com/kainos/ea/dao/EmployeeDao.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.kainos.ea.util.DatabaseConnector.closeConnection;
+import static com.kainos.ea.util.DatabaseConnector.getConnection;
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/EmployeeDao.java:13]: Add extra line after import statements and before class definition for clarity

**Review by hellofadude — APPROVED** [review]
Minor comments. Approved!

**Pull request merged**